docs/framework/data/wcf/streaming-provider-wcf-data-services.md

```diff
@@ -1,124 +1,136 @@
+|<xref:System.Data.Services.Providers.IDataServiceStreamProvider.GetReadStreamUri%2A>|This method is invoked by the data service to return the URI that is used to request the media resource for the media link entry. This value is used to create the `src` attribute in the content element of the media link entry and that is used to request the data stream. When this method returns `null`, the data service automatically determines the URI. Use this method when you need to provide clients with direct access to binary data without using the steam provider.|
+|<xref:System.Data.Services.Providers.IDataServiceStreamProvider.GetStreamContentType%2A>|This method is invoked by the data service to return the Content-Type value of the media resource that is associated with the specified media link entry.|
+|<xref:System.Data.Services.Providers.IDataServiceStreamProvider.GetStreamETag%2A>|This method is invoked by the data service to return the eTag of the data stream that is associated with the specified entity. This method is used when you manage concurrency for the binary data. When this method returns null, the data service does not track concurrency.|
+|<xref:System.Data.Services.Providers.IDataServiceStreamProvider.GetWriteStream%2A>|This method is invoked by the data service to obtain the stream that is used when receiving the stream sent from the client. When you implement <xref:System.Data.Services.Providers.IDataServiceStreamProvider>, you must return a writable stream to which the data service writes received stream data.|
+|<xref:System.Data.Services.Providers.IDataServiceStreamProvider.ResolveType%2A>|Returns a namespace-qualified type name that represents the type that the data service runtime must create for the media link entry that is associated with the data stream for the media resource that is being inserted.|
+
+## Creating the Streaming Data Service
+
+To provide the [!INCLUDE[ssAstoria](../../../../includes/ssastoria-md.md)] runtime with access to the <xref:System.Data.Services.Providers.IDataServiceStreamProvider> implementation, the data service that you create must also implement the <xref:System.IServiceProvider> interface. The following example shows how to implement the <xref:System.IServiceProvider.GetService%2A> method to return an instance of the `PhotoServiceStreamProvider` class that implements <xref:System.Data.Services.Providers.IDataServiceStreamProvider>.
+
+[!code-csharp[Astoria Photo Streaming Service#PhotoServiceStreamingProvider](../../../../samples/snippets/csharp/VS_Snippets_Misc/astoria%20photo%20streaming%20service/cs/photodata.svc.cs#photoservicestreamingprovider)]
```

Comment:
I'm a bit confused.  Shouldn't the space characters change to `_` to match dotnet/samples#810?  

And, that begs the question: will other docs need to be updated for those folder and file name changes?

Reply:
I just saw #11784 

If I understand correctly, we should merge this in the following order:
#11783 
dotnet/samples#810
#11784 

Does that make sense @nschonni ?

Reply:
Thanks, I thought I had undone all of these to handle them separately. I've rebased these ones out so those other PRs should be independent again